src/services/userService.ts

```diff
@@ -130,4 +130,24 @@ export class UserService {
       token,
     };
   }
+
+  static async findUserByID(id, context) {
+    if (!context || !context.user) {
+      throw new UnauthorizedUser({ message: "Usuário não autorizado" });
+    }
```

Comment:
dessa forma já não funcionaria?
```suggestion
    if (!context?.user) {
      throw new UnauthorizedUser({ message: "Usuário não autorizado" });
    }
```